src/shared/bg_pmove.cpp

```diff
@@ -303,7 +303,12 @@ PM_ClipVelocity
 Slide off of the impacting surface
 ==================
 */
-void PM_ClipVelocity( const vec3_t in, const vec3_t normal, vec3_t out )
+static glm::vec3 PM_ClipVelocity( const glm::vec3& in, const glm::vec3& normal )
+{
+	return in - glm::dot( in, normal );
```

Comment:
I think this should be:
```suggestion
	return in - glm::dot( in, normal ) * normal;
```

This could be the reason why you have seen a player going through the ground

Reply:
no, because this is in a different commit, and the bug starts with 643e95d13020836451e271e60f4f74bc74f9383c

Reply:
I didn't check if it cause The Big Bug™ or not, but I'm pretty sure it's a big bug anyway.

Reply:
have you read the code of the original PM_ClipVelocity? This is merely a direct translation of it.

Reply:
The new code looks wrong. Subtracting a scalar from a vector is almost certainly not what you want.

Reply:
indeed, afontain was right on this one, looks like I forgot to add the "* normal" part